### PR TITLE
ENG-1283 Create Test Design for Engineer III

### PR DESCRIPTION
**Pull request by uberflip-jbooth**
Abstract: Create an API that must adhere to 1-3 business rules, e.g. for a library the ISBN must be valid.  Include code, DB design, tests, and comments.  Bonus: create a frontend that uses the API

Potential ideas:

* library
* inventory management
* room scheduler
* employee time clock

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,25 @@ Your task is to create a RESTful API for a time clock system.  All required endp
...
+>
+>(Multiply each digit of the top number by each digit of the bottom number)
+>
+>`130 692 544`
+>
+>`121 212 121`
+>
+>`160 394 584`
+>
+>If you get a 2 digit product, add the digits together. Notice here that 6 * 2 = 12, add 1 and 2 together and get 3.
```
Comment (tamias) [inline README.md]: I think the way this is currently written is a bit confusing.  How about something like this:

Multiple each digit of the SIN by the corresponding digit of the number `121 212 121`.  If you get a 2 digit product, add the digits together.

`130 692 544`

`121 212 121`

`160 394 584`

Notice here that 6 * 2 = 12; add 1 and 2 together and get 3.

File: README.md [hunk 2]
```diff
@@ -29,16 +47,20 @@ Your task is to create a RESTful API for a time clock system.  All required endp
...
+- An employee must be clocked in to be able to be clocked out
+
+### Search Criteria
+
+The list employees endpoint should support searching by first or last name. The timesheet list endpoint should support searching by employee ID and date range.
 
 ## Deliverables
 
-Provide a Github repository with your code, database schema and/or migrations, tests, and API documentation.  Include any setup steps necessary for your interviewer to be able to run your project.  Be prepared to review and explain any design decisions that you made.
+Provide a Github repository with your code, data schema, unit and feature tests, and API documentation.  Include a README with any setup steps necessary for your interviewer to be able to run your project.  Be prepared to screen share a working copy of your API with your interviewer and review and explain the design decisions that you made.
```
Comment (tamias) [inline README.md]: Remove that trailing whitespace.

I'd suggest enabling the option in your IDE to strip trailing whitespace automatically on save.

File: README.md [hunk 3]
```diff
@@ -1 +1,67 @@
...
+- Delete an employee
+- Retrieve an employee
+- Retrieve a list of employees
+- Clock in an employee
+- Clock out an employee
+- Retrieve a list of timesheets
+
+### Constraints
+
+- Employees must have a unique, valid SIN
```
Comment (tamias) [inline README.md]: "Each employee"

**Review by tamias — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (uberflip-jbooth) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (uberflip-jbooth) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (uberflip-jbooth) [reply README.md]: Done

**Pull request merged**